langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AbstractAzureAiSearchEmbeddingStore.java

```diff
@@ -256,29 +263,34 @@ public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedde
     }
 
     @Override
-    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
-        List<Float> vector = referenceEmbedding.vectorAsList();
+    public EmbeddingSearchResult<TextSegment> search(EmbeddingSearchRequest request) {
+
+        List<Float> vector = request.queryEmbedding().vectorAsList();
         VectorizedQuery vectorizedQuery = new VectorizedQuery(vector)
                 .setFields(DEFAULT_FIELD_CONTENT_VECTOR)
-                .setKNearestNeighborsCount(maxResults);
+                .setKNearestNeighborsCount(request.maxResults());
 
         SearchPagedIterable searchResults =
                 searchClient.search(null,
                         new SearchOptions()
+                                .setFilter(filterMapper.map(request.filter()))
                                 .setVectorSearchOptions(new VectorSearchOptions().setQueries(vectorizedQuery)),
                         Context.NONE);
 
         List<EmbeddingMatch<TextSegment>> result = new ArrayList<>();
         for (SearchResult searchResult : searchResults) {
             Double score = fromAzureScoreToRelevanceScore(searchResult.getScore());
-            if (score < minScore) {
+            if (score < request.minScore()) {
                 continue;
             }
             SearchDocument searchDocument = searchResult.getDocument(SearchDocument.class);
             String embeddingId = (String) searchDocument.get(DEFAULT_FIELD_ID);
             List<Double> embeddingList = (List<Double>) searchDocument.get(DEFAULT_FIELD_CONTENT_VECTOR);
-            float[] embeddingArray = doublesListToFloatArray(embeddingList);
-            Embedding embedding = Embedding.from(embeddingArray);
+            Embedding embedding = null;
+            if (embeddingList != null) {
```

Comment:
minor: is it possible that `embeddingList` is `null`?

Reply:
Yes, because the embeddingList is the DEFAULT_FIELD_CONTENT_VECTOR. It is defined obviously "searchable" but not necessarily "retrievable". It's my case, I don't need to retrieve the vector but only the content and metadata.
So in this line : `List<Double> embeddingList = (List<Double>) searchDocument.get(DEFAULT_FIELD_CONTENT_VECTOR);`  embeddingList is null.

Reply:
I am not sure I understand "searchable/retrievable" thing.
I've put a conditional breakpoint on this line and run all the tests and `embeddingList` is always **_not_** null.
Having this extra `if` condition is confusing because it is always `true` (or at least seems to be so from my understanding).